### ISSUE TITLE
Pydantic Field(alias=) is not recognized by Pylint, warns E1123 unexpected-keyword-arg

### ISSUE THREAD

**Issue by bjmc** [pylint-dev/pylint#9090]
### Bug description

I'm not entirely sure if this is a bug, but it's unexpected behavior that I hope could be resolved in Pylint. It might be related to https://github.com/pylint-dev/astroid/pull/1178

I'm using the [dataclasses feature](https://docs.pydantic.dev/latest/usage/dataclasses/) in Pydantic v2.4.2 and I have a dataclass with an aliased field, like so:

`example.py`

```python
from pydantic import Field
from pydantic.dataclasses import dataclass

@dataclass
class Example:
    number: int = Field(alias='n')


example = Example(n=5)
```

This is valid code and runs just fine, but pylint complains with the following

`E1123: Unexpected keyword argument 'n' in constructor call (unexpected-keyword-arg)`

It's maybe worth noting that pylint does *not* complain about the alias usage when the class is a `BaseModel` subclass rather than a dataclass.

```python
from pydantic import Field, BaseModel


class Example2(BaseModel):
    number: int = Field(alias='n')


example = Example2(n=5)
```

### Command used

```shell
pylint example.py
```

### Pylint output

```python
************* Module example
example.py:14:10: E1123: Unexpected keyword argument 'n' in constructor call (unexpected-keyword-arg)
```

### Expected behavior

No warning would be reported because `n` is a valid keyword argument to `__init__()` for the `Example` class.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04.3 LTS

### Additional dependencies

```python
pydantic==2.4.2
```
